File: 74-search-a-2d-matrix/74-search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        m, n = len(matrix), len(matrix[0])
        row, col = 0, 0
        while row < m and col < n:
            if target == matrix[row][col]: return True
            if target > matrix[row][n - 1]: row += 1
            else: col += 1
        return False

File: 74-search-a-2d-matrix/74-search-a-2d-matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m = matrix.length, n = matrix[0].length;
        int row = 0, col = 0;
        while(row < m && col < n) {
            if(target == matrix[row][col]) return true;
            if(target > matrix[row][n - 1]) row++;
            else col++;
        }
        return false;
    }
}